source/nijigenerate/viewport/model/mesheditor/drawable.d

```diff
@@ -204,14 +204,13 @@ public:
     }
 
     override
-    ulong[] getInPolygon(vec2[] points, uint groupId) {
+    ulong[] filterVertices(bool delegate(MeshVertex*) filter) {
         ulong[] matching;
-        foreach(idx, vertex; vertices) {
-            if (!pointInPolygon(vertex, points, groupId)) continue;
-            matching ~= idx;
+        foreach (idx, vertex; mesh.vertices) {
+            if (filter(vertex)) matching ~= idx;
```

Comment:
The current filter implementation seems to have a bug when used with Drawable.

The main issue is the confusion between mesh.vertices and vertices.
This is similar to the problem I encountered previously when trying to switch to getVerticesByIndex:
in some contexts, while getVerticesByIndex may return either mesh.vertices or vertices.

We should clarify the semantics, for example, decide whether these APIs should consistently return deformed vertices or raw mesh vertices, to avoid this ambiguity.

Reply:
thanks for detailed condition. I don't have  environment to confirm now, but I might undertand the situation. I guess that is not related with Deformable, but problem of existing code in inochi-creator where MeshGroups are nested.
If it works correctly in creator, that means this is bug introduced by porting code from inochi2d.

Reply:
This issue was working correctly in the `niji_lasso_selection` branch and also before the refactor to `filterVertices`.  

It seems the problem was introduced during the refactor, not present in the original implementation.

Reply:
Actually, in `source/nijigenerate/viewport/model/mesheditor/drawable.d`, the `getInRect` function uses `vertices()`, which includes deformation, whereas `mesh.vertices` does not include deformation. This difference causes the issue we observed.

Reply:
Thanks for clarification. I fixed it.